src/Stripe.net/Infrastructure/JsonConverters/STJStripeObjectConverter.cs

```diff
@@ -0,0 +1,129 @@
+    /// polymorphic attributes, when the property's type is an interface instead of a concrete type.
+    /// In this case, the converter will use the value of the `object` key in the JSON payload to
+    /// decide which concrete type to instantiate. If no concrete type is found (or if one is found,
+    /// but it's not compatible with the expected interface), then the converter returns `null`.
+    /// </summary>
+    ///
+    public class STJStripeObjectConverter : JsonConverterFactory
+    {
+        /// <summary>
+        /// Determines whether this instance can convert the specified object type.
+        /// </summary>
+        /// <param name="objectType">Type of the object.</param>
+        /// <returns>
+        ///     <c>true</c> if this instance can convert the specified object type; otherwise, <c>false</c>.
+        /// </returns>
+        public override bool CanConvert(Type objectType)
+        {
+            var typeInfo = objectType.GetTypeInfo();
+            return typeInfo.IsInterface && typeInfo.FullName.StartsWith("Stripe");
+        }
+
+        public override JsonConverter CreateConverter(
+            Type type,
+            JsonSerializerOptions options)
+        {
+#pragma warning disable SA1009 // Closing parenthesis should be spaced correctly
+            JsonConverter converter = (JsonConverter)Activator.CreateInstance(
+                typeof(STJStripeObjectConverterInner<>).MakeGenericType(
+                    type),
+                BindingFlags.Instance | BindingFlags.Public,
+                binder: null,
+                args: null,
+                culture: null)!;
+#pragma warning restore SA1009 // Closing parenthesis should be spaced correctly
+
+            return converter;
+        }
+
+        internal class STJStripeObjectConverterInner<T> : JsonConverter<T>
+        where T : class
+        {
+            /// <summary>
+            /// Reads the JSON representation of the object.
+            /// </summary>
+            /// <param name="reader">The <see cref="Utf8JsonReader"/> to read from.</param>
+            /// <param name="typeToConvert">Type of the object.</param>
+            /// <param name="options">The calling serializer's options.</param>
+            /// <returns>The object value.</returns>
+            public override T Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+            {
+                if (reader.TokenType == JsonTokenType.Null)
+                {
+                    return null;
+                }
+
+                var jsonObject = JsonElement.ParseValue(ref reader);
+                var objectValue = jsonObject.GetProperty("object").GetString();
+
+                Type concreteType = StripeTypeRegistry.GetConcreteType(typeToConvert, objectValue);
+                if (concreteType == null)
+                {
+                    // Couldn't find a concrete type to instantiate, return null.
+                    return null;
```

Comment:
Don't we fall back to StripeObject in this case?

Reply:
Nah, in this case we return null.  This part is copied almost verbatim from https://github.com/stripe/stripe-dotnet/blob/4d0c154eb7d4a399d21504f1b9ce1f5be14e8cc6/src/Stripe.net/Infrastructure/JsonConverters/StripeObjectConverter.cs#L58